Mathlib/FieldTheory/SeparablyGenerated.lean

```diff
@@ -0,0 +1,407 @@
+    ∃ (i : ↑(insert x t)), (IsTranscendenceBasis k fun (j : {j // j ≠ i}) ↦ j.1.1) ∧
+      IntermediateField.adjoin k (insert x ↑t : Set K) ≤ (separableClosure
+        (IntermediateField.adjoin k (Subtype.val '' {i}ᶜ)) K).restrictScalars _ := by
+  classical
+  have inst : ExpChar K p := expChar_of_injective_ringHom (algebraMap k K).injective p
+  letI K' := IntermediateField.adjoin k (insert x ↑t : Set K)
+  have : Algebra.IsAlgebraic K' K := by
+    have := ht.isAlgebraic_field
+    rw [IntermediateField.isAlgebraic_adjoin_iff_top,
+      ← AlgebraicIndependent.matroid_spanning_iff] at this ⊢
+    exact .superset this (by simp)
+  letI tx : (insert x ↑t : Set K) → K' := fun a ↦ ⟨a.1, IntermediateField.subset_adjoin _ _ a.2⟩
+  have htx : K'.val '' Set.range tx = insert x ↑t := by
+    rw [← Set.range_comp]
+    exact Subtype.range_val
+  have ⟨i, hi, H⟩ :=
+    exists_isTranscendenceBasis_and_isSeparable_of_linearIndepOn_pow_of_adjoin_eq_top
+    (k := k) (K := K') p hp tx ⟨x, by simp⟩ ?_ ?_ ?_
+  · let K₁ := IntermediateField.adjoin k (tx '' {i}ᶜ)
+    let K₂ := IntermediateField.adjoin k (Subtype.val '' {i}ᶜ)
+    letI e : K₁ ≃ₐ[k] K₂ := (equivMap K₁ K'.val).trans (equivOfEq (by
+      simp only [adjoin_map, coe_val, K₁, K₂, ← Set.image_comp]; rfl))
+    letI := e.toRingHom.toAlgebra
+    have : IsScalarTower K₁ K₂ K := .of_algebraMap_eq' rfl
+    refine ⟨i, hi.algebraMap_comp, ?_⟩
+    intro x hx
+    have := (Algebra.IsSeparable.isSeparable (IntermediateField.adjoin k (tx '' {i}ᶜ)) (K := K')
+      ⟨x, hx⟩).map (L := K) ⟨K'.val.toRingHom, fun _ ↦ by rfl⟩ Subtype.val_injective
+    rw [mem_restrictScalars, mem_separableClosure_iff]
+    exact this.tower_top (IntermediateField.adjoin k (Subtype.val '' {i}ᶜ))
+  · apply IntermediateField.map_injective (IntermediateField.val _)
+    rw [IntermediateField.adjoin_map, ← AlgHom.fieldRange_eq_map, fieldRange_val, htx]
+  · apply IsTranscendenceBasis.of_comp (IntermediateField.val _) Subtype.val_injective
+    convert ht.comp_equiv ?_ using 1; swap
+    · exact ⟨fun x ↦ ⟨x.1, by aesop⟩, fun x ↦ ⟨⟨x.1, by aesop⟩, by aesop⟩,
+        fun x ↦ by aesop, fun x ↦ by aesop⟩
+    ext
+    simp [K', tx]
+  · intro s
+    have := H (s.image (IntermediateField.val _))
+    simp only [coe_val, Finset.coe_image] at this
+    rw [← linearIndepOn_iff_image Subtype.val_injective.injOn] at this
+    rw [linearIndepOn_iff_image (f := (· ^ p)) (frobenius_inj K p).injOn,
+      ← Set.image_comp, ← linearIndepOn_iff_image (f := (· ^ p) ∘ Subtype.val)
+        ((frobenius_inj K p).comp Subtype.val_injective).injOn] at this
+    rw [← LinearMap.linearIndepOn_iff_of_injOn K'.val.toLinearMap Subtype.val_injective.injOn,
+      ← LinearMap.linearIndepOn_iff_of_injOn K'.val.toLinearMap Subtype.val_injective.injOn]
+    exact this
+
+/-- In a finitely generated field extension, there exists a maximal
+separably generated field extension. -/
+lemma exists_isTranscendenceBasis_forall_separableClosure_not_lt
```

Comment:
Instead of allowing t' to be an arbitrary set, we can restrict it to a subset of a fixed generating Finset, right? Then this is easy because there are only finitely many subsets?

Reply:
I removed this lemma in [this commit](https://github.com/leanprover-community/mathlib4/commit/8b4967611f3f3ebb5c5b7898dcbca9591ea3c358).

Reply:
I think this lemma seems useful to have in mathlib. There might be a better statement/proof as you indicated though.

Reply:
Done in [this commit](https://github.com/leanprover-community/mathlib4/commit/2d275d10f112e5c2363d034b41bd577507cf48a7)